constraints.go

```diff
@@ -175,7 +175,7 @@ func parseConstraint(c string) (*constraint, error) {
 	if isX(m[3]) {
 		ver = "0.0.0"
 		dirty = true
-	} else if isX(strings.TrimPrefix(m[4], ".")) {
+	} else if isX(strings.TrimPrefix(m[4], ".")) || m[4] == "" {
```

Comment:
Note (for my future self): what this does it treat a missing minor element as `x` instead of `0`.